samples/sample_fei/src/sample_fei.cpp

```diff
@@ -1297,6 +1297,25 @@ mfxStatus RunPipeline(CEncodingPipeline * pPipeline, mfxU32 session_id)
     return sts;
 }
 
+#define SAMPLE_CHECK_PARSE_RESULT(P, X, ERR) \
+{                                            \
+    if ((X) > (P))                           \
+    {                                        \
+        if (parFile) fclose(parFile);        \
+        return ERR;                          \
+    }                                        \
+}
+
+#define SAMPLE_CHECK_STATUS(X, MSG)          \
+{                                            \
+    if ((X) < MFX_ERR_NONE)                  \
+    {                                        \
+        if (parFile) fclose(parFile);        \
+        MSDK_PRINT_RET_MSG(X, MSG);          \
+        return X;                            \
+    }                                        \
+}
+
```

Comment:
Not good approach to hide side actions inside macro which looks to be only for status check. Could you move file closing to destructor? Another approach is to use fstream, but such change will require some efforts on refactoring

Reply:
Updated with limited scope of "parFile".